CMake/Catch.cmake

```diff
@@ -0,0 +1,175 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
```

Comment:
/me no likey that we include these files. Can't we grab them from a safe location? Catch should either be found with standard CMake mechanisms or included as a submodule, in which case these files should be referenced from the submodule.

Reply:
`catch` is already versioned into `libsonata/tests/contrib`

Reply:
Then I think that should be switched to a git submodule to retain some version information.

Reply:
The author of catch suggests copying the file as was done here: 
https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#where-to-put-it

Reply:
I have seen that copying the catch header file was suggested by the author and most of the people does that, so I thought that adding it as submodule was overkill. That is also why I just  copied the cmake helper files.

But maybe when you start coping several files is worth adding it as a submodule to get always the latest changes.

Reply:
> But maybe when you start coping several files is worth adding it as a submodule to get always the latest changes.

This. Alternatively, I think it would make sense to keep all files from catch in the same folder. Do the versions of the header and the CMake files match?

> The author of catch suggests copying the file as was done here:

That section references a blog post from 2011, I think C++/CMake habits have evolved since then. If we're only using C++, then I think just copying the header is fine. Now we've added CMake files, too.

Reply:
We only need to have one file though; it's in a known place, so it can be directly pointed to.  The version number is embedded into it.

What is the benefit of adding a submodule if there is only a single file?

Reply:
> We only need to have one file though; it's in a known place, so it can be directly pointed to. The version number is embedded into it.
> 
> What is the benefit of adding a submodule if there is only a single file?

The problem is that I was using some cmake helper files to add tests, so I also copied them onto the Cmake folder. I agree with @matz-e that at least we should be consistent with the versions.

Another way would be to add the tests manually like it is done currently in libsonata and remove the need of the cmake files completely.